### PR TITLE
Refactor get topic names and types

### PR DESCRIPTION
**Pull request by wjwwood**
Connects to ros2/ros2#361

### CODE REVIEW

File: rclpy/rclpy/topic_or_service_is_hidden.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+HIDDEN_TOPIC_PREFIX = '_'
+
+
+def topic_or_service_is_hidden(name):
```
Comment (dirk-thomas) [inline rclpy/rclpy/topic_or_service_is_hidden.py]: We should probably update existing code in `ros2cli` to use this new function once merged.

File: rclpy/rclpy/utilities.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+def shutdown():
+    with g_shutdown_lock:
+        return _rclpy.rclpy_shutdown()
+
+
+def try_shutdown():
+    """Shutdown rclpy if not already shutdown."""
+    with g_shutdown_lock:
+        if _rclpy.rclpy_ok():
+            return _rclpy.rclpy_shutdown()
```
Comment (dirk-thomas) [inline rclpy/rclpy/utilities.py]: Should the function return anything in the `else` case (instead of the implicit `None`)?

Comment (wjwwood) [inline rclpy/rclpy/utilities.py]: I don't think so, the shutdown function in the C extension currently returns `None` itself. I could remove the `return` from both shutdown functions here.

Comment (dirk-thomas) [inline rclpy/rclpy/utilities.py]: Than it should be fine as-is.

**Pull request merged**